commit 3f0204b0b7b5ea5fde64ee8197b2d1a281e2fe3a
Author: Florian Fainelli <f.fainelli@gmail.com>
Date:   Thu Aug 24 16:01:13 2017 -0700

    net: systemport: Free DMA coherent descriptors on errors
    
    
    [ Upstream commit c2062ee3d9615828109ffe8089fbf69bed394d05 ]
    
    In case bcm_sysport_init_tx_ring() is not able to allocate ring->cbs, we
    would return with an error, and call bcm_sysport_fini_tx_ring() and it
    would see that ring->cbs is NULL and do nothing. This would leak the
    coherent DMA descriptor area, so we need to free it on error before
    returning.
    
    Reported-by: Eric Dumazet <edumazet@gmail.com>
    Fixes: 80105befdb4b ("net: systemport: add Broadcom SYSTEMPORT Ethernet MAC driver")
    Signed-off-by: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/broadcom/bcmsysport.c b/drivers/net/ethernet/broadcom/bcmsysport.c
index a2c8086c19ca..f328b3d86c13 100644
--- a/drivers/net/ethernet/broadcom/bcmsysport.c
+++ b/drivers/net/ethernet/broadcom/bcmsysport.c
@@ -1342,6 +1342,8 @@ static int bcm_sysport_init_tx_ring(struct bcm_sysport_priv *priv,
 
 	ring->cbs = kcalloc(size, sizeof(struct bcm_sysport_cb), GFP_KERNEL);
 	if (!ring->cbs) {
+		dma_free_coherent(kdev, sizeof(struct dma_desc),
+				  ring->desc_cpu, ring->desc_dma);
 		netif_err(priv, hw, priv->netdev, "CB allocation failed\n");
 		return -ENOMEM;
 	}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1012_3f0204b0b7b5ea5fde64ee8197b2d1a281e2fe3a.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1012_3f0204b0b7b5ea5fde64ee8197b2d1a281e2fe3a.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
